docs/examples/agent-init.md

```diff
@@ -1,5 +1,8 @@
 # Initialize the Agent
 
+We will use `php-http/guzzle6-adapter` and `http-interop/http-factory-guzzle` packages in this example.
```

Comment:
I'm still familiarizing myself with the PHP-HTTP project, but I think I understand how this will work. I believe more clarity is required in the documentation. Please explain why these two packages are being used and when a developer needs to be concerned with this. (If I understand correctly, this is only needed during development when an actual client implementation is not available?)

I think we also need to note that since this package no longer includes an http client implementation, the project must ensure a suitable client is available. What error might the developer see if `HttpClientDiscovery::find()` is unable to load a client and how should that be mitigated?